client/state/jetpack-connect/reducer.js

```diff
@@ -289,10 +295,27 @@ export function jetpackSSOSessions( state = {}, action ) {
 	return state;
 }
 
+export function jetpackConnectSelectedPlans( state = {}, action ) {
+	switch ( action.type ) {
+		case JETPACK_CONNECT_SELECT_PLAN_IN_ADVANCE:
+			const siteSlug = action.site.replace( /^https?:\/\//, '' ).replace( /\//g, '::' );
+			return Object.assign( {}, state, { [ siteSlug ]: action.plan } );
+		case JETPACK_CONNECT_CHECK_URL:
+			return { '*': state[ '*' ] };
+		case JETPACK_CONNECT_COMPLETE_FLOW:
+			return {};
+		case SERIALIZE:
+		case DESERIALIZE:
+			return state;
+	}
+	return state;
+}
+
 export default combineReducers( {
 	jetpackConnectSite,
 	jetpackSSOSessions,
 	jetpackSSO,
 	jetpackConnectAuthorize,
 	jetpackConnectSessions,
+	jetpackConnectSelectedPlans,
```

Comment:
We could use tests for the reducer additions. Could be added in a subsequent PR though.